File: fibonacci/fibonacci.py
#!/usr/bin/python3
import argparse


def fib(n):
    if n <= 1:
        return 1
    return fib(n - 1) + fib(n - 2)


parser = argparse.ArgumentParser(description="Nth Fibonacci")
parser.add_argument("--nth", type=int)
args = parser.parse_args()

print(fib(args.nth if args.nth else input("Enter Nth Fibonacci to compute: ")))

File: fibonacci/fibonacci.nim
import os, strutils

proc fib(n: uint): uint =
    if n <= 1:
        return 1
    return fib(n - 1) + fib(n - 2)

var fibNumber : uint
if paramCount() == 0:
    stdout.write("Enter Nth Fibonacci to compute: ")
    fibNumber = parseUInt(readLine(stdin))
else:
    fibNumber = parseUInt(paramStr(1))

echo(fib(fibNumber))
